exercises/practice/all-your-base/.meta/example.asm

```diff
@@ -0,0 +1,106 @@
+section .text
+global rebase
+
+rebase:
+    ; Argument registers
+    ; %rdi - input digits array
+    ; %esi - number of input digits
+    ; %edx - input base
+    ; %rcx - output array
+    ; %r8d - output base
+
+    xor rax, rax
+
+    ; First, check the bases
+    ; remember - ints are 32 bits!
+    cmp edx, 2
+    jl .bad_base
+
+    cmp r8d, 2
+    jl .bad_base
+
+    ; Part 1 - Parse the number from the input
+    ; %r9d - digit's index
+    ; %eax - sum & result
+    ; %r10 - next digit
+    xor r9, r9
+    xor rax, rax
+
+.parse_digits:
+    xor r10, r10
+
+    cmp r9d, esi
+    je .convert
+
+    ; Load next digit to add
+    mov r10d, dword [rdi + (r9 * 4)]
+    
+    cmp r9d, esi
+    je .convert
```

Comment:
Overall, great work.

On the MIPS track, I gave this exercise a difficulty of 8,   The assembly tracks currently have exceeding few hard exercises (difficulty 8/9/10), and it would nice to work towards having 5+ hard exercises on each track, so people can earn the "Difficult" track trophy.    (See [Java](https://exercism.org/tracks/java/) for an example of a track with 5+ hard exercises.)